# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/storage/StorageObserver.cpp
# Commit: 36aa6faea947
# Full Hash: 36aa6faea9477a13b6d94e396596697e56675018
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-19 03:53:26
# Regressor Bug: 1839918
# File Overlap Count: 6
# Description:
#   Bug 1839918 - Refactor "cookie-changed" notifications to use nsICookieNotification. r=extension-reviewers,cookie-reviewers,sessionstore-reviewers,bvandersloot,edgul,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D185300
# ==============================================================================

diff -r 905592ffa91d -r 36aa6faea947 dom/storage/StorageObserver.cpp
--- a/dom/storage/StorageObserver.cpp	Mon Sep 18 11:25:24 2023 +0000
+++ b/dom/storage/StorageObserver.cpp	Mon Sep 18 11:55:08 2023 +0000
@@ -13,6 +13,8 @@
 #include "StorageUtils.h"
 
 #include "mozilla/BasePrincipal.h"
+#include "nsCOMPtr.h"
+#include "nsICookieNotification.h"
 #include "nsIObserverService.h"
 #include "nsIURI.h"
 #include "nsIPermission.h"
@@ -243,7 +245,11 @@
 
   // Clear everything, caches + database
   if (!strcmp(aTopic, "cookie-changed")) {
-    if (!u"cleared"_ns.Equals(aData)) {
+    nsCOMPtr<nsICookieNotification> notification = do_QueryInterface(aSubject);
+    NS_ENSURE_TRUE(notification, NS_ERROR_FAILURE);
+
+    if (notification->GetAction() !=
+        nsICookieNotification::ALL_COOKIES_CLEARED) {
       return NS_OK;
     }
 